### PR TITLE
Make StatefulSolver non-stochastic and rename StatelessSolver

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Something is wrong with my project setup so I can no longer run the project or the tests easily. However, I was able to hobble together a build and verified a few test runs.

Can you double check by running the tests on your end and spot check the results and runtime of the stateful solver?

### CODE REVIEW

**Review by larkz — APPROVED** [review]
Looks good, took the liberty of changing the naming.

File: app/src/main/kotlin/Reversi/ReversiSolverHeuristicSim.kt [hunk 1]
```diff
@@ -33,11 +33,11 @@ class ReversiSolverHeuristicSim(initialState: ReversiState)
         }
 
         var depth = 0
-        var currentState = node.state
+        var currenStateType = node.state
```
Comment (JunTaoLuo) [inline app/src/main/kotlin/Reversi/ReversiSolverHeuristicSim.kt:36]: This isn't actually the type. This is actually the state.

File: lib/src/main/kotlin/ca/aqtech/mctreesearch4j/GenericSolver.kt [hunk 2]
```diff
@@ -2,7 +2,7 @@ package ca.aqtech.mctreesearch4j
 
 import kotlin.math.max
 
-open class StatelessSolver<StateType, ActionType>(
+open class GenericSolver<StateType, ActionType>(
```
Comment (JunTaoLuo) [inline lib/src/main/kotlin/ca/aqtech/mctreesearch4j/GenericSolver.kt:5]: I like this name 👍

File: app/src/main/kotlin/Reversi/ReversiSolverHeuristicSim.kt (hunk 1 above)
Comment (larkz) [reply app/src/main/kotlin/Reversi/ReversiSolverHeuristicSim.kt:36]: I'm aware now, the global changer on the IDE, messed this up a bit, will quickly hotfix

**Pull request merged**